### DISCUSSION TITLE
How to understand log file?

### DISCUSSION THREAD

**Discussion by user9931** [AdguardTeam/AdGuardHome#4645] [Q&A]
I want to parse adguardhome log (data/querylog.json) with promtail and use it in loki and grafana. Its json file, thats easy to extract values...  One thing that cant understand is what inside Answer key?

For example

{"T":"2022-06-08T13:17:19.347832573+03:00","QH":"www.google.com","QT":"A","QC":"IN","CP":"","Answer":"**Va2BgAABAAYAAAAAA3d3dwZnb29nbGUDY29tAAABAAHADAABAAEAAADeAARA6aJowAwAAQABAAAA3gAEQOmiacAMAAEAAQAAAN4ABEDpomPADAABAAEAAADeAARA6aKTwAwAAQABAAAA3gAEQOmiZ8AMAAEAAQAAAN4ABEDpomo=**","Result":{},"Upstream":"tls://dns.google:853","IP":"192.168.1.8","Elapsed":23886362}
{"T":"2022-06-

**Comment by ainar-g** [comment]
(Note that the query log format is not, and probably will not be, stable.  In particular, see #2290 and the issues that are mentioned there.)

The `Answer` and `OrigAnswer` are both binary representations of the DNS answers received.  If you want to process the query logs, including the answers, you'll need a language with a library capable of decoding the binary DNS message data.
